tools/postReleaseScripts/updateAzureDataStudio.ps1

```diff
@@ -218,33 +218,23 @@ function UpdateGalleryFile
 
     # Create a new PowerShell extension entry
     $powershellEntry = NewPowerShellExtensionEntry -ExtensionVersion $ExtensionVersion
-    $entryStr = ConvertToIndentedJson $powershellEntry -IndentChar "`t" -IndentWidth 1
+    $entryStr = ConvertTo-IndentedJson $powershellEntry -IndentChar "`t" -IndentWidth 1
 
     # Find the position in the existing file where the PowerShell extension should go
     $galleryFileContent = Get-Content -Raw $GalleryFilePath
     $span = FindPSExtensionJsonSpan -GalleryExtensionFileContent $galleryFileContent
-    $startOffset = GetStringOffsetFromSpan -String $galleryFileContent -EndLine $span.Start.Line -Column $span.Start.Column
-    $endOffset = GetStringOffsetFromSpan -String $galleryFileContent -EndLine $span.End.Line -StartLine $span.Start.Line -Column $span.End.Column -InitialOffset $startOffset
+    $startOffset = Get-StringOffsetFromSpan -String $galleryFileContent -EndLine $span.Start.Line -Column $span.Start.Column
+    $endOffset = Get-StringOffsetFromSpan -String $galleryFileContent -EndLine $span.End.Line -StartLine $span.Start.Line -Column $span.End.Column -InitialOffset $startOffset
 
     # Create the new file contents with the inserted segment
-    $newGalleryFileContent = ReplaceStringSegment -String $galleryFileContent -NewSegment $entryStr -StartIndex $startOffset -EndIndex ($endOffset+1) -AutoIndent
+    $newGalleryFileContent = New-StringWithSegment -String $galleryFileContent -NewSegment $entryStr -StartIndex $startOffset -EndIndex ($endOffset+1) -AutoIndent
 
     # Write out the new entry
-    SetFileContent $GalleryFilePath $newGalleryFileContent
+    Set-Content -Path $GalleryFilePath -Value $newGalleryFileContent -Encoding utf8NoBOM -NoNewline
```

Comment:
I think this means it'll only work on PS6 (`utf8noBOM`) - which I'm ok with for this script, but wanted to call this out

Reply:
Yeah, that's right

Reply:
Basically the semver handling means this will only work in PS 6, so with that in mind I've gone with a few PS6-specific things, since we can always opt for pwsh in AzDO (and unlike building, it's not important to be able to run release code in WinPS-only contexts)

Reply:
Should probably add this then:

```
#Requires -Version 6
```